test/api_test.py

```diff
@@ -21,12 +25,6 @@ def test_purge_by_key(self):
         self.api.authenticate_by_key(self.api_key)
         self.assertTrue(self.api.purge_key(self.service_id, 'foo'))
 
-    def test_cookie_purge_by_key(self):
-        self.api.deauthenticate()
-        self.api.authenticate_by_password(self.user, self.password)
-        with self.assertRaises(fastly.AuthenticationError):
-            self.api.purge_key(self.service_id, 'foo')
```

Comment:
Why was it necessary to remove this test?

Reply:
My mistake. I had thought Fastly removed username/password authentication but re-reviewing the docs I see that is not the case! I will bring these 3 tests back.

Reply:
@psbanka Revisiting this... it doesn't seem like authenticating by password is supported by Fastly's API anymore. In the `auth.py` file there's a call to the `/login` endpoint: https://github.com/fastly/fastly-py/blob/master/fastly/auth.py#L18

but in the API docs I cannot find any mention of a `/login` endpoint: https://docs.fastly.com/api/auth

The reason I removed these 3 tests is because I thought Fastly deprecated username+password authentication in favor of Key-based authentication. Is this not the case?

Reply:
Ah, okay. You are correct in principle: actually fastly does still support login for now, but it is deprecated and we want folks to stop using it. 

But to remove the tests without getting rid of the rest of the deprecated functionality (e.g. https://github.com/fastly/fastly-py/blob/master/fastly/auth.py#L17) is the wrong move to make in my opinion because It's only half-done. So, for _this_ pull-request, let's keep the tests in because all we're trying to do is support python 3. 

I'll follow-up with a second PR which pulls out all the username/password functionality (and tests!) and do a major version-bump for that PR. Does that sound okay?

Reply:
Yep I agree with that! I'll add the tests back in (even though they'll be failing).